### PR TITLE
feat(padded-describes): add the padded-describes rule

### PR DESCRIPTION
**Pull request by robinjoseph08**
### What

Create the `padded-describes` rule.

Creating a plugin is the way to create custom rules and then this new plugin will be required from https://github.com/lob/eslint-config-lob and then added to the config to be applied everywhere.

Most of this code is ripped from the [`padded-blocks`](https://github.com/eslint/eslint/blob/master/lib/rules/padded-blocks.js) rule in the core eslint repo. My main contribution is the `isDescribeBlock` function.

It was kinda hard to wrap my head around all of the AST node stuff, but the main helper for me to make sure I was doing the right thing was looking at the tests. If you have any additional test cases you want me to add, just let me know.

@mgartner

### CODE REVIEW

**Comment by mgartner** [comment]
Ahh i think you mean padded-blocks in your link above. will this also require spaces between it blocks, or should we write another rule for that?

**Comment by robinjoseph08** [comment]
You're right, updated the description.

I thought we didn't want to require spaces between `it` blocks, right? We only want them for `describe`s?

**Comment by mgartner** [comment]
i meant between it blocks not inside. to prevent this:

    it('asdf' ...
      //....
    })
    it('asdf'......

**Comment by robinjoseph08** [comment]
Ahh gotcha. Lemme look into it and see if it makes sense to add it into this rule.

**Comment by robinjoseph08** [comment]
From the looks of it, it should be a separate rule to enforce that space.

File: lib/rules/padded-describes.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+/**
+ * Checks if the given non empty block node is the callback of a describe function.
+ * @param {ASTNode} node - the AST node of a BlockStatement
+ * @returns {Boolean} whether or not the block is the callback of a describe function
+ */
+function isDescribeBlock (node) {
+  return isFunctionExpression(node.parent.type) &&
+    node.parent.parent.type === 'CallExpression' &&
+    node.parent.parent.callee.name === 'describe';
```
Comment (mgartner) [inline lib/rules/padded-describes.js]: :money_with_wings:

File: lib/rules/padded-describes.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    if (lastComment && isLocatedBefore(lastToken, lastComment)) {
+      lastLine = lastComment.loc.end.line;
+    }
+
+    return lastLine <= expectedLastLine;
+  }
+
+  return {
+    BlockStatement: (node) => {
```
Comment (mgartner) [inline lib/rules/padded-describes.js]: Should this be `DescribeBlockStatement`? How does changing this key affect things?

**Comment by mgartner** [comment]
LGTM just 1 question.

File: lib/rules/padded-describes.js (hunk 2 above)
Comment (robinjoseph08) [inline lib/rules/padded-describes.js]: Yeah, it needs to be `BlockStatement` because that designates the type of AST node this function will be called on.

**Pull request merged**